src/query/Query.ts

```diff
@@ -455,9 +453,19 @@ export class Query<M extends Model = Model> {
         return
       }
 
-      model[key] = isArray(relatedSchema)
-        ? this.newQueryForRelation(attr).reviveMany(relatedSchema)
-        : this.newQueryForRelation(attr).reviveOne(relatedSchema)
+      // Inverse polymorphic relations have the same parent and child model so we need to query using the type stored
+      // in the parent model
+      if (attr instanceof MorphTo) {
+        const relatedType = model[attr.$getType() as string]
+
+        model[key] = isArray(relatedSchema)
+          ? this.newQuery(relatedType).reviveMany(relatedSchema)
+          : this.newQuery(relatedType).reviveOne(relatedSchema)
```

Comment:
For inverse polymorphic relations the child model set to the same model as the parent as we do not know the child entity ahead of time. As a result we cannot rely on the parent / child relation to create a query.